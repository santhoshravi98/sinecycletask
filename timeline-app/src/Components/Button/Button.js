import React from "react";
import ButtonCss from "../Button/Button.module.css";
const Button = (props) => {
    return (
        <button data-testid="button_test"
            disabled={props.disabled}
            onClick={props.buttonClickMethod}
            className={[ButtonCss.Button, ButtonCss[props.buttonType]].join(" ")}
        >
            {props.children}
        </button>
    );
};
export default Button;
