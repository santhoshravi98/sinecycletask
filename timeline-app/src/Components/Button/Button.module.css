.Button {
    background-color: #096dd9;
    border: none;
    color: white;
    outline: none;
    cursor: pointer;
    font: inherit;
    padding: 10px;
    margin: 10px;
    font-weight: bold;
}

.Button:first-of-type {
    margin-left: 48%;
    padding-left: 5px;
}

.Success {
    color: white;
}

.Danger {
    color: #944317;
}

.Button:disabled{
    color: gray;
    cursor:not-allowed;
}