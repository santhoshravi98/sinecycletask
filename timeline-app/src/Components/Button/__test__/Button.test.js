import React from 'react'
import ReactDOM from 'react-dom'
import {render} from '@testing-library/react'
import Button from '../Button'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import {fireEvent, cleanup} from '@testing-library/react';

afterEach(cleanup)
it("renders without crashing",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<div></div>,div); 
})

it("renders Button Container correctly",()=>{
    const {queryByTestId} = render(<Button></Button>)
    expect(queryByTestId("button_test")).toBeTruthy();
})

it("matches snapshot",() => {
const tree = renderer.create(<Button></Button>).toJSON();
expect(tree).toMatchSnapshot();
})

// it("has correct class",() => {
//     const  = render(<TimeLine></TimeLine>)
//     expect(getByTestId("ul-list").
//     })
