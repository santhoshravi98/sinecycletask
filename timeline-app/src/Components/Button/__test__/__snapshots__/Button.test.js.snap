// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="Button undefined"
  data-testid="button_test"
/>
`;
